SuperSaaS/Client.py

```diff
@@ -31,6 +31,27 @@
 API_VERSION = '2'
 VERSION = '1.2.2'
 
+import time
+
+
+
+
+class RateLimiter:
+    # The rate limiter allows a maximum of 4 requests to be made within the specified time window, which is defined by the WINDOW_SIZE constant
+    WINDOW_SIZE = 1
+    MAX_PER_WINDOW = 4
+
+    def __init__(self):
+        self.queue = [None] * self.MAX_PER_WINDOW
+
+    def throttle(self):
+        # Represents the timestamp of the oldest request within the time window
+        oldest_request = self.queue.pop() if self.queue else None
+        # Push the current timestamp into the queue
+        self.queue.append(time.time())
+        # This ensures that the client does not make requests faster than the defined rate limit
+        if oldest_request and (time_elapsed := time.time() - oldest_request) < self.WINDOW_SIZE:
+            time.sleep(self.WINDOW_SIZE - time_elapsed)
```

Comment:
Sleep blocks but we don't seem to have other async operations so should be fine.